lib/io.toit

```diff
@@ -734,6 +734,117 @@ class ByteArrayReader_ extends Reader:
   close_ -> none:
     data_ = null
 
+
+interface OutStrategy:
+  /**
+  Writes the given $data to this writer.
+
+  Returns the number of bytes written.
+
+  See $Writer.try-write_.
+  */
+  // This is a protected method. It should not be "private".
+  try-write_ data/Data from/int to/int -> int
+
+  /**
+  Closes this writer.
+
+  See $Writer.close_.
+  */
+  // This is a protected method. It should not be "private".
+  close-writer_ -> none
+
+class Out_ extends Writer:
+  strategy_/OutStrategy
+
+  constructor .strategy_:
+
+  try-write_ data/Data from/int to/int -> int:
+    return strategy_.try-write_ data from to
+
+  close_ -> none:
+    strategy_.close-writer_
+
+abstract mixin OutMixin implements OutStrategy:
+  out_/Out_? := null
+
+  out -> Writer:
+    if not out_: out_ = Out_ this
```

Comment:
Clearly this is just me, but I'd avoid reading the field twice.

Reply:
done.